tinylog-impl/src/main/java/org/tinylog/pattern/ThreadContextToken.java

```diff
@@ -58,21 +68,39 @@ public Collection<LogEntryValue> getRequiredLogEntryValues() {
 
 	@Override
 	public void render(final LogEntry logEntry, final StringBuilder builder) {
-		String value = logEntry.getContext().get(key);
-		if (value == null) {
-			builder.append(defaultValue);
+		if (key != null) {
+			String value = logEntry.getContext().get(key);
+			if (value == null) {
+				builder.append(defaultValue);
+			} else {
+				builder.append(value);
+			}
 		} else {
-			builder.append(value);
+			boolean first = true;
+			for (Map.Entry<String, String> contextEntry : logEntry.getContext().entrySet()) {
+				if (first) {
+					first = false;
+				} else {
+					builder.append(DELIMITER);
+				}
+				builder.append(contextEntry.getKey()).append(SEPARATOR).append(contextEntry.getValue());
+			}
 		}
 	}
 
 	@Override
 	public void apply(final LogEntry logEntry, final PreparedStatement statement, final int index) throws SQLException {
-		String value = logEntry.getContext().get(key);
-		if (value == null && !DEFAULT_EMPTY_VALUE.equals(defaultValue)) {
-			statement.setString(index, defaultValue);
+		if (key == null || key.isEmpty()) {
```

Comment:
Can you add a JUnit test for this case? Currently line 93 - 96 are not covered by any test.

Reply:
Done!